packages/foundry/contracts/other/notional/NotionalJoin.sol

```diff
@@ -137,32 +116,24 @@ contract NotionalJoin is IJoin, ERC1155TokenReceiver, AccessControl() {
     }
 
     /// @dev Transfer `amount` `asset` to `user`
-    function _exit(address user, uint128 amount)
-        internal
-        beforeMaturity
-        returns (uint128)
-    {
+    /// @param user Address of receiver of fCash tokens
+    /// @param amount Amount of ERC1155 tokens
+    function _exit(address user, uint128 amount) internal beforeMaturity returns (uint128) {
```

Comment:
Redundant `beforeMaturity` check

Reply:
Right, because `exit` has checks if `block.timestamp < maturity`, before executing `_exit`

```solidity
    function exit(address user, uint128 amount) external override auth returns (uint128) {
        if (block.timestamp < maturity) {
            return _exit(user, amount);
        } else {
            if (accrual == 0) redeem(); // Redeem all fCash, switch to underlying join, set accrual.
            return _exitUnderlying(user, uint256(amount).wmul(accrual).u128());
        }
    }
```

Reply:
`beforeMaturity`  removed.